Crash with OutOfBoundsException

**Describe the bug**
Lespas crashes or never finishes loading when opening it.

**To Reproduce**
Steps to reproduce the behavior:
To just stall
1. Open app
2. enter url of nextcloud server
3. put in username and password, hit login button and watch it not do anything
To crash
1. use ZXing qr code scanner and app password to mostly skip the login process
2. lets me choose a home folder on the nextcloud
3. watch the app crash

**Expected behavior**
open the folder overview

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Tablet**
 - Device: Samsung Galaxy Tab S6 lite
 - OS: Android 13
 - Version: 2.9.9 of lespas
 - has an sd card slot that is not currently in use

**Additional context**
have built it locally, will add logcat below, and tried the version from google play, crashed as well

<details>
<summary>logcat entry</summary>

```
FATAL EXCEPTION: main
Process: site.leos.apps.lespas, PID: 19969
java.lang.RuntimeException: Unable to start activity ComponentInfo{site.leos.apps.lespas/site.leos.apps.lespas.MainActivity}: java.lang.RuntimeException: Cannot create an instance of class site.leos.apps.lespas.sync.ActionViewModel
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4184)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4340)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2584)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:226)
	at android.os.Looper.loop(Looper.java:313)
	at android.app.ActivityThread.main(ActivityThread.java:8810)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:604)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
Caused by: java.lang.RuntimeException: Cannot create an instance of class site.leos.apps.lespas.sync.ActionViewModel
	at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.kt:318)
	at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.kt:302)
	at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.kt:276)
	at androidx.lifecycle.SavedStateViewModelFactory.create(SavedStateViewModelFactory.kt:128)
	at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.kt:184)
	at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.kt:150)
	at androidx.lifecycle.ViewModelLazy.getValue(ViewModelLazy.kt:53)
	at androidx.lifecycle.ViewModelLazy.getValue(ViewModelLazy.kt:35)
	at site.leos.apps.lespas.MainActivity.getActionsPendingModel(MainActivity.kt:69)
	at site.leos.apps.lespas.MainActivity.onCreate(MainActivity.kt:172)
	at android.app.Activity.performCreate(Activity.java:8657)
	at android.app.Activity.performCreate(Activity.java:8636)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1417)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4165)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4340) 
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101) 
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2584) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:226) 
	at android.os.Looper.loop(Looper.java:313) 
	at android.app.ActivityThread.main(ActivityThread.java:8810) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:604) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067) 
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Constructor.newInstance0(Native Method)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
	at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.kt:310)
	at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.kt:302) 
	at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.kt:276) 
	at androidx.lifecycle.SavedStateViewModelFactory.create(SavedStateViewModelFactory.kt:128) 
	at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.kt:184) 
	at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.kt:150) 
	at androidx.lifecycle.ViewModelLazy.getValue(ViewModelLazy.kt:53) 
	at androidx.lifecycle.ViewModelLazy.getValue(ViewModelLazy.kt:35) 
	at site.leos.apps.lespas.MainActivity.getActionsPendingModel(MainActivity.kt:69) 
	at site.leos.apps.lespas.MainActivity.onCreate(MainActivity.kt:172) 
	at android.app.Activity.performCreate(Activity.java:8657) 
	at android.app.Activity.performCreate(Activity.java:8636) 
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1417) 
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4165) 
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4340) 
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101) 
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2584) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:226) 
	at android.os.Looper.loop(Looper.java:313) 
	at android.app.ActivityThread.main(ActivityThread.java:8810) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:604) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067) 
Caused by: java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
	at site.leos.apps.lespas.helper.Tools.getLocalRoot(Tools.kt:554)
	at site.leos.apps.lespas.sync.ActionViewModel.<init>(ActionViewModel.kt:42)
	at java.lang.reflect.Constructor.newInstance0(Native Method) 
	at java.lang.reflect.Constructor.newInstance(Constructor.java:343) 
	at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.kt:310) 
	at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.kt:302) 
	at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.kt:276) 
	at androidx.lifecycle.SavedStateViewModelFactory.create(SavedStateViewModelFactory.kt:128) 
	at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.kt:184) 
	at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.kt:150) 
	at androidx.lifecycle.ViewModelLazy.getValue(ViewModelLazy.kt:53) 
	at androidx.lifecycle.ViewModelLazy.getValue(ViewModelLazy.kt:35) 
	at site.leos.apps.lespas.MainActivity.getActionsPendingModel(MainActivity.kt:69) 
	at site.leos.apps.lespas.MainActivity.onCreate(MainActivity.kt:172) 
	at android.app.Activity.performCreate(Activity.java:8657) 
	at android.app.Activity.performCreate(Activity.java:8636) 
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1417) 
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4165) 
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4340) 
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101) 
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2584) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:226) 
	at android.os.Looper.loop(Looper.java:313) 
	at android.app.ActivityThread.main(ActivityThread.java:8810) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:604) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```

</details>

using the debugger, I could see that `PreferenceManager.getDefaultSharedPreferences(context).getBoolean(SettingsFragment.KEY_STORAGE_LOCATION, true)` is false, but the `context.getExternalFilesDirs(null)` contains just a single entry `/storage/emulated/0/Android/data/site.leos.apps.lespas/files`

--- a/app/src/main/java/site/leos/apps/lespas/auth/NCSelectHomeFragment.kt
+++ b/app/src/main/java/site/leos/apps/lespas/auth/NCSelectHomeFragment.kt
@@ -286,8 +286,12 @@ class NCSelectHomeFragment: Fragment() {
                                 getString(R.string.chinese_map_pref_key),
                                 getString(R.string.nearby_convergence_pref_key),
                                 getString(R.string.remove_meta_data_before_sharing_pref_key),
-                                getString(R.string.use_low_resolution_to_share_pref_key),
-                                SettingsFragment.KEY_STORAGE_LOCATION, -> editor.putBoolean(key, value.toBoolean())
+                                getString(R.string.use_low_resolution_to_share_pref_key) -> editor.putBoolean(key, value.toBoolean())
+                                // only put false here when SD card is actually present
+                                SettingsFragment.KEY_STORAGE_LOCATION -> {
+                                    editor.putBoolean(key, value.toBoolean() || (requireContext().getExternalFilesDirs(null).size < 2))
+                                }
+
 
                                 // String preferences
                                 getString(R.string.auto_theme_perf_key),

